brainiak/utils/simulator.py

```diff
@@ -0,0 +1,1058 @@
+    """Generate the signal dependent noise
+    This noise depends on things like the signal or the timing of the
+    experiment.
+
+    Parameters
+    ----------
+
+    stimfunction : 1 Dimensional array
+        This is the timecourse of the stimuli in this experiment
+
+    tr_duration : int
+        How long is a TR, in seconds
+
+    z_score : bool
+        Is the data to be normalized
+
+    motion_sigma : float
+
+        How much noise is left over after pre-processing has been
+        done. This is noise specifically on the task events
+
+    drift_sigma : float
+
+        What is the sigma on the distribution that coefficients are
+        randomly sampled from
+
+    auto_reg_sigma : float
+
+
+    auto_reg_order : float
+
+
+    auto_reg_rho : float
+
+
+    physiological_sigma : float
+
+        How variable is the signal as a result of physiology,
+        like heart beat and breathing
+
+    Returns
+    ----------
+    one dimensional array, float
+        Generates the temporal noise timecourse for these parameters
+
+
+        """
+
+    # Set up common parameters
+    # How many TRs are there
+    trs = len(stimfunction)
+
+    # What time points are sampled by a TR?
+    timepoints = list(range(0, trs * tr_duration))[::tr_duration]
+
+    # Make each noise type
+    noise_task = _generate_noise_temporal_task(stimfunction,
+                                               motion_sigma,
+                                               )
+
+    noise_drift = _generate_noise_temporal_drift(trs,
+                                                 tr_duration,
+                                                 drift_sigma,
+                                                 timepoints,
+                                                 )
+
+    noise_phys = _generate_noise_temporal_phys(timepoints,
+                                               physiological_sigma,
+                                               )
+
+    noise_autoregression = _generate_noise_temporal_autoregression(
+        auto_reg_rho, auto_reg_order, auto_seg_sigma, timepoints, )
+
+    # Do you want to z score it?
+    if z_score == 1:
+        noise_task = _z_score(noise_task)
+        noise_phys = _z_score(noise_phys)
+        noise_drift = _z_score(noise_drift)
+        noise_autoregression = _z_score(noise_autoregression)
+
+    # add the noise (it would have been nice to just add all of them in a
+    # single line but this was causing formatting problems)
+    noise_temporal = noise_task
+    noise_temporal = noise_temporal + noise_phys
+    noise_temporal = noise_temporal + noise_drift
+    noise_temporal = noise_temporal + noise_autoregression
+
+    return noise_temporal
+
+
+def _generate_noise_spatial(dimensions,
+                            sigma=-4.0,
+                            ):
+    """Generate code for Gaussian Random Fields. Based on code from
+    http://andrewwalker.github.io/statefultransitions/post/gaussian-fields/
```

Comment:
Not sure how to find the license. I used the code from this blog as a base but there is no license with the blog as far as I can tell.